asdf/tests/test_asdf.py

```diff
@@ -363,3 +362,27 @@ def test_reading_extension_metadata():
         buff = yaml_to_asdf(content)
         with assert_no_warnings():
             open_asdf(buff)
+
+
+def test_bad_input(tmp_path):
+    """Make sure these functions behave properly with bad input"""
+    text_file = str(tmp_path / "test.txt")
+
+    with open(text_file, "w") as fh:
+        fh.write("I <3 ASDF!!!!!")
+
+    with pytest.raises(ValueError):
+        open_asdf(text_file)
+
+
+def test_unclosed_file(tmp_path):
```

Comment:
@WilliamJamieson Here's the regression test. I also moved test_bad_input here from test_fits_embed as it did not test fits.

Reply:
Excellent! Looks good.